blocks/page_list/templates/itinerary/AddWalkToListDialog.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+export default class AddWalkToListDialog extends React.Component {
+  constructor(...args) {
+    super(...args);
+    this.state = {
+      newList: null
+    }
+  }
+
+  propTypes() {
+    return {
+      lists: React.PropTypes.array,
+      add: React.PropTypes.func,
+      createList: React.PropTypes.func,
+      walkSelected: React.PropTypes.number,
+    }
+  }
+
+  render() {
+    const {lists, add, createList, activeWalk} = this.props;
+    const {newList} = this.state;
+    //selectedWalk comes from where
+
+    const allLists = lists.map(list => {
```

Comment:
This could probably be a stateless component instead, e.g. `const allLists = lists.map(list => <List value={list} />);`